src/canvas-extractor/tests/graphql/extractor/test_extractor_integration.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+import pytest
+
+from pandas import DataFrame
+from pathlib import Path
+from sqlalchemy import create_engine
+
+from edfi_canvas_extractor.graphql.extractor import GraphQLExtractor
+from edfi_canvas_extractor.graphql.schema import query_builder
+
+from edfi_canvas_extractor.graphql.assignments import assignments_synced_as_df
+from edfi_canvas_extractor.graphql.courses import courses_synced_as_df
+from edfi_canvas_extractor.graphql.enrollments import enrollments_synced_as_df
+from edfi_canvas_extractor.graphql.sections import sections_synced_as_df
+from edfi_canvas_extractor.graphql.students import students_synced_as_df
+
+
+@pytest.fixture(scope="class")
+def gql_class():
+    CANVAS_BASE_URL = os.environ['CANVAS_BASE_URL']
+    CANVAS_ACCESS_TOKEN = os.environ['CANVAS_ACCESS_TOKEN']
+    START_DATE = "2021-01-01"
+    END_DATE = "2030-01-01"
+
+    gql = GraphQLExtractor(
+        CANVAS_BASE_URL,
+        CANVAS_ACCESS_TOKEN,
+        "1",
+        START_DATE,
+        END_DATE,
+        )
+    yield gql
+
+
+@pytest.fixture(scope="class")
+def graphql_class(gql_class):
```

Comment:
SHOULD: We should move towards having only a single fetch request for the entire test class instead of passing in this fixture to each test function in the class. We can make use of the concept of [autouse fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request) and [request fixture](https://docs.pytest.org/en/stable/reference/reference.html#request) to achieve this. Something like this:
```
@pytest.fixture(autouse=True, scope="class")
def fetch_graphql_data(request, gql_run):
    request.cls._graphql_extraction = gql_run
```
The tests can then use assertions as follows:
```
...
def test_gql_data_fetched(self):
        assert self._graphql_extraction.has_data is True

def test_gql_courses_not_empty(self):
        courses = self._graphql_extraction.get_courses()
        assert courses is not None
        assert isinstance(courses, list)
...
```